keras_cv/models/resnet.py

```diff
@@ -0,0 +1,537 @@
+            the 4D tensor output of the
+            last convolutional layer.
+        - `avg` means that global average pooling
+            will be applied to the output of the
+            last convolutional layer, and thus
+            the output of the model will be a 2D tensor.
+        - `max` means that global max pooling will
+            be applied.
+      num_classes: optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified.
+      classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+      **kwargs: For backwards compatibility only.
+
+        Returns:
+      A `keras.Model` instance.
+    """
+    if kwargs:
+        raise ValueError("Unknown argument(s): %s" % (kwargs,))
+    if weights and not tf.io.gfile.exists(weights):
+        raise ValueError(
+            "The `weights` argument should be either `None` or the path to the "
+            "weights file to be loaded. Weights file not found at location: {weights}"
+        )
+
+    if include_top and not num_classes:
+        raise ValueError(
+            "If `include_top` is True, you should specify `num_classes`. "
+            f"Received: num_classes={num_classes}"
+        )
+
+    img_input = layers.Input(shape=input_shape)
+
+    if include_rescaling:
+        img_input = layers.Rescaling(1 / 255.0)(img_input)
+
+    x = layers.Conv2D(
+        64, 7, strides=2, use_bias=True, padding="same", name="conv1_conv"
+    )(img_input)
+
+    if not preact:
+        x = layers.BatchNormalization(axis=BN_AXIS, epsilon=1.001e-5, name="conv1_bn")(
+            x
+        )
+        x = layers.Activation("relu", name="conv1_relu")(x)
+
+    x = layers.MaxPooling2D(3, strides=2, padding="same", name="pool1_pool")(x)
+
+    x = stack_fn(x)
+
+    if preact:
+        x = layers.BatchNormalization(axis=BN_AXIS, epsilon=1.001e-5, name="post_bn")(x)
+        x = layers.Activation("relu", name="post_relu")(x)
+
+    if include_top:
+        x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+        x = layers.Dense(
+            num_classes, activation=classifier_activation, name="predictions"
+        )(x)
+    else:
+        if pooling == "avg":
+            x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+        elif pooling == "max":
+            x = layers.GlobalMaxPooling2D(name="max_pool")(x)
+
+    inputs = img_input
+
+    # Create model.
+    model = tf.keras.Model(inputs, x, name=name)
+
+    if weights is not None:
+        model.load_weights(weights)
+
+    return model
+
+
+# v1 block
+def V1Block(filters, kernel_size=3, stride=1, conv_shortcut=True, name=None):
+    """A residual block.
```

Comment:
Would it make sense to add a shot description of how the v1 block differs from the v2 block?

Reply:
+1